### PR TITLE
Feature/HEXXLE-114 Tile Hand Implementation

### PR DESCRIPTION
**Pull request by denadex**

### CODE REVIEW

File: unity/Assets/Scripts/Logic/Hand.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
+﻿using Hexxle.Interfaces;
+using Hexxle.Unity;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Hexxle.Logic
+{
+    public class Hand
+    {
+        private List<ITile> Slots;
```
Comment (RicardoMonteiroSimoes) [inline unity/Assets/Scripts/Logic/Hand.cs:10]: Any reason for capitalized variables? I thought the convention said any non-contant variables use camelCase

**Review by RicardoMonteiroSimoes — APPROVED** [review]

File: unity/Assets/Scripts/Logic/Hand.cs (hunk 1 above)
Comment (denadex) [reply unity/Assets/Scripts/Logic/Hand.cs]: I completely mixed that up..., should be fixed now

File: unity/Assets/Scripts/Logic/Hand.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            int counter = 0;
+            for(int i = 0; i < handSize; i++)
+            {
+                if(slots[i] == null)
+                {
+                    counter++;
+                }
+            }
+            return counter;
+        }
```
Comment (DennisBannerman) [inline unity/Assets/Scripts/Logic/Hand.cs:80]: You can use c#'s LINQ to write this a little smoother. [LINQ](https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.count?view=net-5.0#System_Linq_Enumerable_Count__1_System_Collections_Generic_IEnumerable___0__System_Func___0_System_Boolean__)

Would probably look something like 
``` csharp
public int EmptySlots()
{
    return slots.Count(s => s == null);
}
```

**Review by DennisBannerman — APPROVED** [review]
Looks good. Lots of changed stuff, but we already talked about that and I think, that's something that's a "problem" for the whole codebase and we might just need to schedule some time to refactor in a later sprint.

Added a comment about LINQ. Personally, I find it one of the best features of C# over Java - Java has Streams since some time, but it's not as accessible and easy as LINQ is. Have a look, if you're interested.

**Pull request merged**